subiquity/server/server.py

```diff
@@ -787,8 +787,13 @@ async def _extract_autoinstall_from_cloud_config(
             # of the schema validation error, otherwise continue
 
             # Filter only the bad keys
+            # Intentionally (key, None) ! Have a good plan (and reason) for
+            # getting the data of a key. It is not guaranteed that the
+            # offending keys to be top-level (or in?) in the combined config.
+            # LP: #2062988
+
             potential_autoinstall: dict[str, Any] = dict(
```

Comment:
I guess this should be a set instead

Reply:
It needs to be a dict to pass to `filter_autoinstall`. We could re-work `filter_autoinstall` to take either a dict or a set but I feel like that starts to make it overly complex.